flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/operators/multipleinput/TableOperatorWrapperTest.java

```diff
@@ -51,26 +48,23 @@ public void testBasicInfo() {
         wrapper3.addInput(wrapper1, 1);
         wrapper3.addInput(wrapper2, 2);
 
-        assertTrue(wrapper1.getInputEdges().isEmpty());
-        assertTrue(wrapper1.getInputWrappers().isEmpty());
-        assertEquals(Collections.singletonList(wrapper3), wrapper1.getOutputWrappers());
-        assertEquals(
-                Collections.singletonList(new Edge(wrapper1, wrapper3, 1)),
-                wrapper1.getOutputEdges());
-
-        assertTrue(wrapper2.getInputEdges().isEmpty());
-        assertTrue(wrapper2.getInputWrappers().isEmpty());
-        assertEquals(Collections.singletonList(wrapper3), wrapper2.getOutputWrappers());
-        assertEquals(
-                Collections.singletonList(new Edge(wrapper2, wrapper3, 2)),
-                wrapper2.getOutputEdges());
-
-        assertTrue(wrapper3.getOutputEdges().isEmpty());
-        assertTrue(wrapper3.getOutputWrappers().isEmpty());
-        assertEquals(Arrays.asList(wrapper1, wrapper2), wrapper3.getInputWrappers());
-        assertEquals(
-                Arrays.asList(new Edge(wrapper1, wrapper3, 1), new Edge(wrapper2, wrapper3, 2)),
-                wrapper3.getInputEdges());
+        assertThat(wrapper1.getInputEdges().isEmpty()).isTrue();
+        assertThat(wrapper1.getInputWrappers().isEmpty()).isTrue();
+        assertThat(wrapper1.getOutputWrappers()).containsExactly(wrapper3);
+        assertThat(wrapper1.getOutputEdges()).containsExactly(new Edge(wrapper1, wrapper3, 1));
+
+        assertThat(wrapper2.getInputEdges().isEmpty()).isTrue();
+        assertThat(wrapper2.getInputWrappers().isEmpty()).isTrue();
```

Comment:
```suggestion
        assertThat(wrapper2.getInputEdges()).isEmpty();
        assertThat(wrapper2.getInputWrappers()).isEmpty();
```
hmm, I thought this migration worked already.

Reply:
It works, but sometimes the type inference hangs and the tool fallbacks to something like what you see

Reply:
Actually, `isEmpty` is not implemented :smile: There are just too many ways of checking if a collection is empty or not